Assets/Scripts/PlayerControls.cs

```diff
@@ -38,123 +48,212 @@ void Update()
+			     player.stats.CanUseStat( Stat.Stamina ) )
 			{
-				_actorPhysics.JumpCheck();
+				player.physics.ClimbSurface( player.controls.GetMoveInput() );
 			}
-
-			if ( _holdButton )
+			else
 			{
-				_actorPhysics.ClimbCheck();
+				player.physics.ChangeState( ActorStates.Falling );
 			}
+		}
+
+		public override void Exit()
+		{
+			player.animator.SetBool( "isClimbing", false );
+			player.stats.StopUsingStat( Stat.Stamina );
+			player.physics.StopClimbing();
+		}
+	}
 
-			if ( _useButton.down )
+	public class Grounded : PhysicsState
+	{
+		PlayerActor player;
+
+		public Grounded( PlayerActor player )
+		{
+			this.player = player;
+		}
+
+		public override void Enter() { }
+
+		public override void Update()
+		{
+			if ( player.physics.GroundedCheck() )
 			{
-				// This allows us to do one raycast for both actions
-				// which is good since we do RaycastAll(), which is expensive.
-				RaycastHit hitInfo;
-				if ( RaycastForward( out hitInfo ) )
+				if ( player.controls.jumpButton.down &&
+				     player.physics.JumpCheck() )
 				{
-					if ( !_cutting.CutCheck( hitInfo ) )
+					player.physics.ChangeState( ActorStates.Jumping );
+				}
+
+				if ( player.controls.holdButton &&
+				     player.physics.ClimbCheck() )
+				{
+					player.physics.ChangeState( ActorStates.Climbing );
+				}
+
+				if ( player.controls.useButton.down )
+				{
+					if ( player.inventory.CanUseItemWithoutTarget() )
 					{
-						_actor.actorResources.CheckUseItem( hitInfo );
+						player.inventory.UseItem();
+					}
+					else
+					{
+						// This allows us to do one raycast for both actions
+						// which is good since we do RaycastAll(), which is expensive.
+						RaycastHit hitInfo;
+						if ( player.controls.RaycastForward( out hitInfo ) )
+						{
+							if ( !player.cutting.CutCheck( hitInfo ) )
+							{
+								player.inventory.UseItemWithTarget( hitInfo );
+							}
+						}
 					}
 				}
-			}
 
-			_actorPhysics.RollCheck();
+				player.controls.GroundMovement();
 
-			GroundMovement();
+				player.animator.SetFloat( "moveSpeed", player.rigidbody.velocity.magnitude );
+			}
+			else
+			{
+				player.physics.ChangeState( ActorStates.Falling );
+			}
 		}
+
+		public override void Exit() { }
 	}
 
-	void Gliding()
+	public class Gliding : PhysicsState
 	{
-		if ( _jumpButton.down )
+		PlayerActor player;
+
+		public Gliding( PlayerActor player )
 		{
-			_actorPhysics.JumpCheck();
+			this.player = player;
 		}
 
-		if ( _holdButton )
+		public override void Enter()
 		{
-			_actorPhysics.ClimbCheck();
+			player.physics.StartGliding();
+			player.stats.StartUsingStat( Stat.Gliding );
+			player.animator.SetBool( "isGliding", true );
 		}
 
-		if ( !_actorPhysics.GroundedCheck() &&
-		     _holdButton &&
-		     _actor.actorStats.CanUseStat( Stat.Gliding ) )
+		public override void Update()
 		{
-			_actorPhysics.GlideMovement( GetMoveDirection() );
+			if ( player.controls.jumpButton.down &&
```

Comment:
This is like `Input.GetButtonDown()`, so it's only true if the button was just pressed this frame. I'm not sure how we can make this more clear while sticking the the Unity standards.